servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -492,12 +485,15 @@ private ImmutableLogEntry commitToLogEntry(boolean fetchAll, Commit commit) {
   private static CommitMeta enhanceCommitMeta(
       Hash hash, CommitMeta commitMeta, List<Hash> additionalParents) {
     ImmutableCommitMeta.Builder updatedCommitMeta = commitMeta.toBuilder().hash(hash.asString());
-    if (additionalParents != null && additionalParents.size() == 1) {
-      // Only add the 1st commit ID. The MERGE_PARENT_PROPERTY was introduced for compatibility
-      // with older clients. There is currently only one use case for the property: exposing the
-      // commit ID of the merged commit.
-      updatedCommitMeta.putProperties(
-          CommitMeta.MERGE_PARENT_PROPERTY, additionalParents.get(0).asString());
+    if (additionalParents != null && !additionalParents.isEmpty()) {
+      additionalParents.forEach(p -> updatedCommitMeta.addParentCommitHashes(p.asString()));
```

Comment:
I think we should add the "proper" parent as the first element of `ParentCommitHashes` and "additional" parents as the following elements. That was the idea behind `CommitMeta.getParentCommitHashes()` (essentially following `git` approach).

Essentially, the "proper" parent of the main lineage is always `getParentCommitHashes().get(0)` and is always present (except for the "root" commit, perhaps). `getParentCommitHashes().get(1)` is the merge parent (in one-way merge cases), `.get(2)` is the second merge parent (no existing use cases for this ATM), etc.

Reply:
makes sense